flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusReporter.java

```diff
@@ -50,11 +47,7 @@ int getPort() {
         return port;
     }
 
-    @Override
-    public void open(MetricConfig config) {
-        super.open(config);
-
-        String portsConfig = config.getString(ARG_PORT, DEFAULT_PORT);
+    PrometheusReporter(@Nullable final String portsConfig) {
```

Comment:
This is not nullable, which we should also enforce via `Preconditions.checkNotNull`